.github/workflows/unit-tests.yaml

```diff
@@ -2,8 +2,6 @@ name: Unit Tests
 on: [push, pull_request]
```

Comment:
Maybe we should just call it 'Tests' instead of 'Unit tests' as this 
is really combination of:  unit & integrational (integration, functional, e2e). 

Does github offers any form of stats that allow to track flakiness per workflow ? 
This would give some motivation to have actually 4 workflows.

Reply:
Rename make sense.

As for flakyness tracking I didn't find anything provided by GitHub. There are some external tools like https://github.com/GoogleCloudPlatform/flaky-service which tracks flakyness on per test level (not workflow).